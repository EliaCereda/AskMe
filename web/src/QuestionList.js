import React, {Component} from "react";

class QuestionList extends Component {
    render() {
        const questionList = this.props.questions.map((question) =>
            <div className="QuestionItem" key={question.id}>
                <Question {...question} />
                <form onSubmit={this.handleSubmit}>
                    <input type="submit" value="Read answers"/>
                </form>
            </div>
        );

        // Display the most recent questions first.
        questionList.reverse();

        return questionList;
    }
}

export default QuestionList;

class Question extends Component {
    render() {
        return (
            <article className="Question">
                <header>
                    <h3>{this.props.title}</h3>
                    <h4>opened by {this.props.user}</h4>
                </header>

                <p>{this.props.body}</p>
            </article>
        );
    }
}
