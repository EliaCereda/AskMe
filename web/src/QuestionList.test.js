import React from 'react';
import QuestionList from "./QuestionList";

describe('QuestionList', () => {
    it('renders correctly', () => {
        const questions = [
            {
                id: 1,
                title: 'titolo della domanda',
                body: 'corpo della domanda',
                user: 'utente',
                _links: {self: 'questions/1'}
            },
            {
                id: 2,
                title: 'titolo di un\'altra domanda',
                body: 'corpo di un\'altra domanda',
                user: 'anonimo',
                _links: {self: 'questions/2'}
            },
            {
                id: 3,
                title: 'un titolo',
                body: 'un po di testo',
                user: 'qualcuno',
                _links: {self: 'questions/3'}
            },
        ];

        const wrapper = shallow(<QuestionList questions={questions} />);
        expect(wrapper).toMatchSnapshot();
    });
});
