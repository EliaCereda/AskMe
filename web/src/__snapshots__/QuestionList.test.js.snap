// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionList renders correctly 1`] = `
Array [
  <div
    className="QuestionItem"
    key="2"
  >
    <Question
      _links={
        Object {
          "self": "questions/3",
        }
      }
      body="un po di testo"
      id={2}
      title="un titolo"
      user="qualcuno"
    />
    <Link
      replace={false}
      to="questions/3"
    >
      <button>
        Read answers
      </button>
    </Link>
  </div>,
  <div
    className="QuestionItem"
    key="2"
  >
    <Question
      _links={
        Object {
          "self": "questions/2",
        }
      }
      body="corpo di un'altra domanda"
      id={2}
      title="titolo di un'altra domanda"
      user="anonimo"
    />
    <Link
      replace={false}
      to="questions/2"
    >
      <button>
        Read answers
      </button>
    </Link>
  </div>,
  <div
    className="QuestionItem"
    key="1"
  >
    <Question
      _links={
        Object {
          "self": "questions/1",
        }
      }
      body="corpo della domanda"
      id={1}
      title="titolo della domanda"
      user="utente"
    />
    <Link
      replace={false}
      to="questions/1"
    >
      <button>
        Read answers
      </button>
    </Link>
  </div>,
]
`;
